src/apps/lwaftr/lwaftr.lua

```diff
@@ -332,9 +349,33 @@ local function ipv6_encapsulate(lwstate, pkt, next_hdr_type, ipv6_src, ipv6_dst,
       print("encapsulated packet:")
       lwdebug.print_pkt(pkt)
    end
+
    return transmit(lwstate.o6, pkt)
 end
 
+local function enqueue_encapsulation(lwstate, pkt, ipv4, port)
+   -- TODO: Switch to streaming lookup interface.
+   local ipv6_dst, ipv6_src = binding_lookup_ipv4(lwstate, ipv4, port)
+   if not ipv6_dst then
+      -- Lookup failed.
+      if debug then print("lookup failed") end
+      if lwstate.policy_icmpv4_outgoing == lwconf.policies['DROP'] then
+         return drop(pkt)
+      elseif get_ipv4_proto(get_ethernet_payload(pkt)) == proto_icmp then
+         -- FIXME: Why don't we send dst/host unreachable for ICMP?
+         return drop(pkt)
```

Comment:
It's not me :)  The existing code effectively has two encapsulation paths, one through ICMP and one for everything else.  See below, line 372 in the original code.  This check here is to preserve the original behavior.

That said, the reason you would want to do this is to prevent ICMP error messages to unreachable hosts from generating ICMP error mesages to unreachable hosts.  We can look at changing the behavior in another patch, but it will need tests.